src/viur/core/bones/phone.py

```diff
@@ -0,0 +1,119 @@
+            apply_default_country_code (bool): Whether to apply the default country code if none is provided.
+            **kwargs (Any): Additional keyword arguments.
+        """
+        self.custom_regex: Pattern[str] = custom_regex or re.compile(
+            r'^(?:\+|00)?[1-9]\d{0,2}[-\s]?\d{1,4}[-\s]?\d{1,4}[-\s]?\d{1,4}$|'
+            r'^(0\d{1,4}[-\s]?\d{1,4}[-\s]?\d{1,4})$'
+        )
+        self.phone_regex: Pattern[str] = (
+            re.compile(self.custom_regex) if isinstance(self.custom_regex, str) else self.custom_regex
+        )
+        self.default_country_code: str = default_country_code
+        self.apply_default_country_code: bool = apply_default_country_code
+
+        super().__init__(max_length=max_length, **kwargs)
+
+    def isInvalid(self, value: str) -> Optional[str]:
+        """
+        Checks if the provided phone number is valid or not.
+
+        Args:
+            value (str): The phone number to be validated.
+
+        Returns:
+            Optional[str]: An error message if the phone number is invalid or None if it is valid.
+
+        The method checks if the provided phone number is valid according to the following criteria:
+        1. The phone number must not be empty.
+        2. The phone number must match the provided or default phone number format.
+        3. If the phone number has no country code and apply_default_country_code is True,
+           the default country code is applied.
+        """
+        if not value:
+            return "No value entered"
+
+        if not self.phone_regex.match(value):
+            return "Invalid phone number entered"
+
+        is_invalid: Optional[str] = super().isInvalid(value)
+
+        if is_invalid:
+            return is_invalid
+        return None
+
+    def singleValueFromClient(
+        self, value: str, skel: Any, bone_name: str, client_data: Any
+    ) -> Tuple[Optional[str], Optional[List[ReadFromClientError]]]:
+        """
+        Processes a single value from the client, applying the default country code if necessary and validating the
+        phone number.
+
+        Args:
+            value (str): The phone number provided by the client.
+            skel (Any): Skeleton data (not used in this method).
+            bone_name (str): The name of the bone (not used in this method).
+            client_data (Any): Additional client data (not used in this method).
+
+        Returns:
+            Tuple[Optional[str], Optional[List[ReadFromClientError]]]:
+            A tuple containing the processed phone number and an optional list of errors.
+        """
+        # Replace country code starting with 00 with +
+        if value.startswith('00'):
+            value = '+' + value[2:]
+
+        # Apply default country code if none is provided and apply_default_country_code is True
+        if self.apply_default_country_code and not value.startswith(('+', '00')):
```

Comment:
When `value` previously started with "00", it is replaced by "+". The test for "00" again here makes no sense.
```suggestion
        if self.default_country_code and value[0] != "+":
```